core/domain/skill_services.py

```diff
@@ -777,10 +778,9 @@ def delete_skill(committer_id, skill_id, force_deletion=False):
             still retained in the datastore. This last option is the preferred
             one.
     """
-    skill_model = skill_models.SkillModel.get(skill_id)
-    skill_model.delete(
-        committer_id, feconf.COMMIT_MESSAGE_SKILL_DELETED,
-        force_deletion=force_deletion)
+    transaction_services.run_in_transaction(
+        skill_models.SkillModel.delete_multi, [skill_id],
+        committer_id, '', force_deletion=force_deletion)
```

Comment:
You don't need to run this in transaction.

Reply:
Done.